src/lib/viewers/BaseViewer.js

```diff
@@ -195,6 +200,14 @@ class BaseViewer extends EventEmitter {
         }
     }
 
+    /**
+     * Checks to see whether the viewer has been setup already
+     * @return {boolean} - Boolean whether the viewer has been set up yet
+     */
+    isSetup() {
+        return this.containerEl && this.containerEl.classList.contains(CLASS_BOX_PREVIEW_HAS_VIEWER);
```

Comment:
May want to use an instance property, if possible, so we can avoid tracking this state via the DOM.

Reply:
Moved to instance property as you suggested. Also decoupled the `setup()` from `load()` and invoke it separately. Couldn't do it in the constructor because it interferes with the `prefetch` logic